### ISSUE TITLE
`test_split::test_round_robin_limited_file_descriptors` fails sporadically on `x86_64-unknown-linux-musl` 

### ISSUE THREAD

**Issue by cre4ture** [uutils/coreutils#6031]
Example failure:

```
failures:

---- test_split::test_round_robin_limited_file_descriptors stdout ----
run: /target/x86_64-unknown-linux-musl/debug/coreutils split -n r/40 onehundredlines.txt
thread 'test_split::test_round_robin_limited_file_descriptors' panicked at 'Command was expected to succeed. code: 1
stdout = 
 stderr = split: unable to open 'xbj'; aborting
', tests/by-util/test_split.rs:1646:10
stack backtrace:
   0: rust_begin_unwind
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/panicking.rs:578:5
   1: core::panicking::panic_fmt
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/panicking.rs:67:14
   2: tests::common::util::CmdResult::success
             at ./tests/common/util.rs:400:9
   3: tests::common::util::UCommand::succeeds
             at ./tests/common/util.rs:1599:9
   4: tests::test_split::test_round_robin_limited_file_descriptors
             at ./tests/by-util/test_split.rs:1643:5
   5: tests::test_split::test_round_robin_limited_file_descriptors::{{closure}}
             at ./tests/by-util/test_split.rs:1642:48
   6: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    test_split::test_round_robin_limited_file_descriptors
```

As far as my analyses is correct, the filename contains also the chunk number as its automatically generated:
xaa - 1
xab - 2
xac - 3
...
xaz - 26
xba - 27
...
xbj - 36

So it seems like the limited fd handling works in general as otherwise it would not be able to generate 36 files, but only 9.

I'm currently trying to fix the issue just by making the handling of the fd limitation more robust.

Other example links:
https://github.com/cre4ture/coreutils/actions/runs/8103387718/job/22147977331?pr=9

Here it was 100% successfull despite 12500 runs, but it was executed as single test only.
Probably the missing background-load caused stability.
https://github.com/cre4ture/coreutils/actions/runs/8122922623/job/22202872494?pr=9

**Comment by cre4ture** [comment]
UPDATE: I found a way to reproduce the issue. It includes quite some effort, but its all scripted.
I'm using 25 Github runners, each executing all out tests in a loop of 500 iterations.
This gives a guaruantee to have multiple cases where this test fails.

https://github.com/cre4ture/coreutils/actions/runs/8123936859/job/22204881082?pr=9

```
failures:

---- test_split::test_round_robin_limited_file_descriptors stdout ----
run: /target/x86_64-unknown-linux-musl/debug/coreutils split -n r/40 onehundredlines.txt
thread 'test_split::test_round_robin_limited_file_descriptors' panicked at 'Command was expected to succeed.
stdout = 
 stderr = split: Instantiation of writer failed due to error: Custom { kind: Other, error: "unable to re-open 'xaa'; aborting" }. Existing writer number: 34
', tests/by-util/test_split.rs:1647:10
stack backtrace:
   0: rust_begin_unwind
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/panicking.rs:578:5
   1: core::panicking::panic_fmt
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/panicking.rs:67:14
   2: tests::common::util::CmdResult::success
             at ./tests/common/util.rs:388:9
   3: tests::common::util::UCommand::succeeds
             at ./tests/common/util.rs:1586:9
   4: tests::test_split::test_round_robin_limited_file_descriptors
             at ./tests/by-util/test_split.rs:1644:5
   5: tests::test_split::test_round_robin_limited_file_descriptors::{{closure}}
             at ./tests/by-util/test_split.rs:1643:48
   6: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    test_split::test_round_robin_limited_file_descriptors

test result: FAILED. 3044 passed; 1 failed; 30 ignored; 0 measured; 0 filtered out; finished in 88.85s
```


Additionally, I extended the logs to trace down one suspicion that I had.
`stderr = split: Instantiation of writer failed due to error: Custom { kind: Other, error: "unable to re-open 'xaa'; aborting" }. Existing writer number: 34` 
This statement shows that the process was actually creating much more readers than the allowed 9 ones from the test limit.

The code in the testframework gives an explanation: It applies the limits to the process id of the already started child process. This means the child process had already some execution time before the limit is applied.
This is what we can see in the log. It was creating 34 writers before the limit was applied. Then, suddenly its not able to open more files. It tries to close one file and still fails, because its still over the limit.

**Comment by BenWiederhake** [comment]
Was this fixed by #6043? I haven't seen this failure around in a while.
